src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Action;
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransactions(List<String> dateList) {
+        List<FruitTransaction> result = new ArrayList<>();
+        for (String string : dateList) {
+            String[] stringArray = string.split(",");
+            Action action = Action.fromCode(stringArray[0]);
+            String fruit = stringArray[1];
+            int quantity = Integer.parseInt(stringArray[2]);
+
+            result.add(new FruitTransaction(action, fruit, quantity));
```

Comment:
It's important to validate the parts of the `stringArray` before using them. Ensure that the array has the expected number of elements and that each element is in the expected format. This will prevent `ArrayIndexOutOfBoundsException` or `NumberFormatException`.